app/javascript/components/Payment/Payment.js

```diff
@@ -200,13 +202,15 @@ export class Payment extends Component {
   }
 
   emitTransactionSubmitted() {
+    const userId = Cookie.get('__bpmx');
     const eventPayload = {
+      user_id: userId,
```

Comment:
If we're saving `member_id` into the `__bpmx` cookie, won't this be the same as `champaign.personalization.member.id`? Or is this intended for new members.

Reply:
@eyko 
__bpmx cookie value is same as champaign.personalization.member.id 

In many pages post the user registration the  `champaign.personalization.member.id` does not get the updated value until we refresh the page.

To overcome this issue I have introduced the cookie.